src/components/Board.js

```diff
@@ -8,17 +8,37 @@ const generateSquareComponents = (squares, onClickCallback) => {
   // Complete this for Wave 1
   // squares is a 2D Array, but 
   //  you need to return a 1D array
-  //  of square components
+  //  of 9 square components
 
-}
+  let squaresFlat = [];
+  for (let i = 0; i < squares.length; i++) {
+    squaresFlat = squaresFlat.concat(squares[i]);
+  };
+
+  console.log(squaresFlat);
+
+  const squareComponents = squaresFlat.map((square) => {
+    return (
+        <Square 
+          key={square.id}
+          id={square.id}
+          value={square.value}
+          onClickCallback={onClickCallback}
+        />
+    );
+  });
+
+  return squareComponents;
```

Comment:
So funnily enough JavaScript has a built in `flat` method:

```suggestion
  const squaresFlat = squares.flat();

  const squareComponents = squaresFlat.map((square) => {
    return (
        <Square 
          key={square.id}
          id={square.id}
          value={square.value}
          onClickCallback={onClickCallback}
        />
    );
  });

  return squareComponents;
```

It even has a method that combines a `flat` and a `map`:

```suggestion
  const squareComponents = squares.flatMap((square) => {
    return (
        <Square 
          key={square.id}
          id={square.id}
          value={square.value}
          onClickCallback={onClickCallback}
        />
    );
  });

  return squareComponents;
```